lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -1126,7 +1126,7 @@ findFieldWithName(const std::vector<swift::reflection::FieldInfo> &fields,
     if (name != field.Name) {
       // A nonnull TypeRef is required for enum cases, where it represents cases
       // that have a payload. In other types it will be true anyway.
-      if (field.TR)
+      if (!is_enum || field.TR)
```

Comment:
Would be good to comment (I assume it's because enums have exactly 1 field?)
Also: should this be `!is_enum &&` field.TR`?
Finally: testcase?

Reply:
> Would be good to comment (I assume it's because enums have exactly 1 field?)

I think the comment there still applies. If it's an enum and there's no typeref then we don't increment the index, because that's a case without a payload.


> Also: should this be !is_enum && field.TR`?

I don't think so? If it's not not an enum we always want to increment the index (given the previous comment), in case it's an enum we wan't to increment only if there's a typeref.

> Finally: testcase?

If we merge https://github.com/apple/swift/pull/61081 `TestSwiftFoundation.py` will catch this. Otherwise I'll have to figure out a way to trigger this behavior.

Reply:
Ah right.